### DISCUSSION TITLE
Copilot only seems to do anything if you put your code in a function.

### DISCUSSION THREAD

**Discussion by Xyphyn** [community/community#6685] [Copilot Conversations]
For some reason, when programming with copilot, copilot only gives you _useful_ suggestions if you put it in a function
For example, if you write this:
```python
'''Get the user's favorite colors and store each one of them in a list'''

```
Copilot doesn't appear to be doing anything.
Although if you put it in a function,
```python
def get_favorite_colors():
    '''Get the user's favorite colors and store each one of them in a list'''
    fav_colors = []
    while True:
        color = input("Enter a color: ")
        if color == "":
            break
        fav_colors.append(color)
    return fav_colors -Copilot
```
Copilot helps you with it. I find this weird, as not all code is put inside of functions.

**Comment by williamcotton** [comment]
Oh man, I was wondering how to do this! Yes, fix this issue as it is entirely not obvious that this is how to get results from the triple single quote blocks!
